pkg/eks/fargate.go

```diff
@@ -60,10 +61,33 @@ func (t *fargateProfilesTask) Do(errCh chan error) error {
 	return nil
 }
 
+// Check if target Fargate Profile existence
+func isTargetFargateProfileExists(target string, existedProfileNames []string, clusterName string) bool {
+	for _, existedProfileName := range existedProfileNames {
+		if target == existedProfileName {
+			logger.Info("Fargate profile %q already exists on EKS cluster %q", target, clusterName)
+			return true
+		}
+	}
+	return false
+}
+
 // DoCreateFargateProfiles creates fargate profiles as specified in the config
 func DoCreateFargateProfiles(ctx context.Context, config *api.ClusterConfig, fargateClient FargateClient) error {
 	clusterName := config.Metadata.Name
+
+	// Get existed farget profile list before entering loop
+	existedProfileNames, err := fargateClient.ListProfiles(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed to get Fargate Profile list")
+	}
```

Comment:
To avoid too many `ListFargateProfiles` API calls, leave it outside the loop.

Reply:
**_Discussion:_**

Another approach would be [DescribeFargateProfile](https://docs.aws.amazon.com/eks/latest/APIReference/API_DescribeFargateProfile.html) with error detection in `isTargetFargateProfileExists()`?

```
// if fargateProfile not exist, it should returns `ResourceNotFoundException`
func tryDescribeFargateProfile(...) error { ... }

func isTargetFargateProfileExists(target, clusterName string) bool {
    if err := tryDescribeFargateProfile(...); err != nil {
        return true
    }
    return false
}
```

I personally prefer `ListFargateProfiles` with simple string compare. What do you think?

Reply:
`fargateClient.ListProfiles` approach is fine 👍🏻 we tend to avoid making too many API calls.